lib/schema_plus/core/active_record/schema_dumper.rb

```diff
@@ -68,17 +68,36 @@ def table(table, _)
               env.table.trailer = m[:trailer].split("\n").map(&:strip).reject{|s| s.blank?}
               table_objects = m[:columns].strip.split("\n").map { |col|
                 cs = col.strip
+                result = nil
+                # first match expression index case
                 m = cs.match %r{
-                ^
-                t\.(?<type>\S+) \s*
-                  [:'"](?<name>[^"\s]+)[,"]? \s*
-                  ,? \s*
-                  (?<options>.*)
-                $
-                }x
-                if !m.nil?
-                  SchemaDump::Table::Column.new name: m[:name], type: m[:type], options: eval("{" + m[:options] + "}"), comments: []
-                else
+                  ^
+                  t\.index \s*
```

Comment:
I must say that this is rather complex. It's the first time I'm seeing this, can you explain what this does?

Reply:
So the issue is a matter of priority handling..  What this code is doing is parsing the output of the Rails Schema dumper.. Previously it was only handling 2 cases.. the "general" case, and the special case of `t.index [:col_a, :col_b], name: 'index_name'`.   However I need to handle the case of an expression index which is different in Rails 5.x before the general case..

Reply:
Hmm, the over-engineer in me would say that regex is a hack, but writing a full blown SQL parser is hard, too. Let's refactor to use the multiple cases and then think over it again in future,